### ISSUE TITLE
Web UI cannot securely communicate with API without loopback address in certificate SANs

### ISSUE THREAD

**Issue by cwjohnston** [sensu/sensu-go#2458]
## Expected Behavior

It should be possible to use the web UI over https when the backend is configured with a certificate that does not contain `127.0.0.1` and/or `::` in the subject alternative names (SANs).

## Current Behavior

Under current implementation, Sensu Go dashboard communicates with the API via `127.0.0.1` or `::` loopback addresses. If these addresses are not in the certificate's SANs, the web UI will generate an error when attempting a graphQL query.

## Steps to Reproduce (for bugs)
1. follow the cfssl steps from our clustering guide at https://docs.sensu.io/sensu-go/5.0/guides/clustering/
2. configure the backend to use the resulting certificate for the web UI
3. attempt to use the web UI and encounter a stack trace (included below)

## Context

Some environments or compliance regimes will require the use of certificates signed by a PKI managed by a third party -- e.g. different team in the organization, or a globally recognized CA.

Including loopback addresses in SANs to work around this may contravene recommended certificate issuing practices and cause problems for those needing to obtaining certs from external CAs.

**Web UI Stack Trace:**

at `new ApolloError` (/static/js/main.1cf95e3f.js:7)
at `undefined` (/static/js/main.1cf95e3f.js:60)
at `undefined` (/static/js/main.1cf95e3f.js:60)
at `undefined` (/static/js/main.1cf95e3f.js:60)
at `QueryManager.broadcastQueries` (/static/js/main.1cf95e3f.js:60)
at `Object.next` (/static/js/main.1cf95e3f.js:60)
at `notifySubscription` (/static/js/main.1cf95e3f.js:60)
at `onNotify` (/static/js/main.1cf95e3f.js:60)
at `SubscriptionObserver.next` (/static/js/main.1cf95e3f.js:60)
at `undefined` (/static/js/main.1cf95e3f.js:36)
at `Object.next` (/static/js/main.1cf95e3f.js:36)
at `notifySubscription` (/static/js/main.1cf95e3f.js:60)
at `onNotify` (/static/js/main.1cf95e3f.js:60)

## Your Environment

* Sensu version used (sensuctl, sensu-backend, and/or sensu-agent): `1ff7092`
* Installation method (packages, binaries, docker etc.):
* Operating System and version (e.g. Ubuntu 14.04):

**Comment by jamesdphillips** [comment]
@cwjohnston there is probably something obvious I am missing, but, I attempted the following steps and was unable to replicate.

```
$ echo '{"CN":"CA","key":{"algo":"rsa","size":2048}}' | cfssl gencert -initca - | cfssljson -bare ca -
$ echo '{"signing":{"default":{"expiry":"43800h","usages":["signing","key encipherment","server auth","client auth"]}}}' > ca-config.json

$ export ADDRESS=backend-1,backend-2
$ export NAME=backend-1
$ echo '{"CN":"'$NAME'","hosts":[""],"key":{"algo":"rsa","size":2048}}' | cfssl gencert -config=ca-config.json -ca=ca.pem -ca-key=ca-key.pem -hostname="$ADDRESS" -profile=peer - | cfssljson -bare $NAME

$ ls
backend-1-key.pem backend-1.csr     backend-1.pem     ca-config.json    ca-key.pem        ca.csr            ca.pem

$ ./bin/sensu-backend start --cert-file backend-1.pem --key-file backend-1-key.pem --trusted-ca-file ca.pem --api-host 'backend-1' --dashboard-host 'backend-1' -d /tmp/sensu --log-level debug
```

Would you be able to send your configuration?

**Comment by jamesdphillips** [comment]
@cwjohnston would you be able to confirm if https://github.com/sensu/sensu-go/pull/2463 fixes the issue you encountered?

**Comment by cwjohnston** [comment]
@jamesdphillips I'm not sure if its #2463 or another change, but this issue seems to be resolved as of 5.0.1.
